compiler/src/iree/compiler/Codegen/Common/CombineLayoutTransformation.cpp

```diff
@@ -198,12 +210,159 @@ foldExtractSliceIntoMapScatter(RewriterBase &rewriter,
+  SmallVector<OpFoldResult> lbs(ubs.size(), rewriter.getIndexAttr(0));
+  SmallVector<DistributionConfig> distConfigs = distributionConfigFn(
+      padOp.getSourceType().getShape(), rewriter.getContext());
+
+  auto innerLoopBuilder = [&](OpBuilder &b, Location loopLoc, ValueRange ivs) {
+    // We need to scatter the padding values according to the existing
+    // mapScatterOp transformation, so clone the transformation into the
+    // loop nest.
+    auto clonedMapScatterOp = cast<MapScatterOp>(b.clone(*mapScatterOp));
+    Block &clonedTransformBody =
+        clonedMapScatterOp.getTransformationRegion().getBlocks().front();
+    // Get a pointer to the YieldOp before inlining the Block.
+    auto yieldOp =
+        cast<IREE::LinalgExt::YieldOp>(clonedTransformBody.getTerminator());
+    rewriter.inlineBlockBefore(&clonedTransformBody, clonedMapScatterOp, ivs);
+    rewriter.eraseOp(clonedMapScatterOp);
+    OpBuilder::InsertionGuard g(b);
+    b.setInsertionPointAfter(yieldOp);
+    // Compute the indices into the outputBuffer, and the if condition to
+    // write the padding values. Padding values must obey the existing mask
+    // of the current mapScatterOp, and also be in the low or high pad
+    // range of the padOp.
+    SmallVector<OpFoldResult> low = padOp.getMixedLowPad();
+    Value writeCond = nullptr;
+    for (auto [l, srcSize, idx] : llvm::zip_equal(low, padSrcSizes, ivs)) {
+      Value lowVal = getValueOrCreateConstantIndexOp(rewriter, loc, l);
+      auto isLowPad = rewriter
+                          .create<arith::CmpIOp>(loc, arith::CmpIPredicate::ult,
+                                                 idx, lowVal)
+                          ->getResult(0);
+      Value highPadStart = getValueOrCreateConstantIndexOp(
+          rewriter, loc, IREE::LinalgExt::addOfrs(b, loopLoc, l, srcSize));
+      auto isHighPad =
+          rewriter
+              .create<arith::CmpIOp>(loc, arith::CmpIPredicate::uge, idx,
+                                     highPadStart)
+              ->getResult(0);
+      Value isPad = rewriter.create<arith::OrIOp>(loc, isLowPad, isHighPad);
+      if (!writeCond) {
+        writeCond = isPad;
+        continue;
+      }
+      writeCond = rewriter.create<arith::OrIOp>(loc, writeCond, isPad);
+    }
+    SmallVector<Value> storeIndices(yieldOp.getOperands());
+    rewriter.eraseOp(yieldOp);
+    Value mask = storeIndices.pop_back_val();
+    writeCond = rewriter.create<arith::AndIOp>(loc, writeCond, mask);
+    // Create the store to the outputBuffer.
+    auto thenBuilder = [&](OpBuilder &nestedBuilder, Location ifLoc) {
+      nestedBuilder.create<memref::StoreOp>(
+          ifLoc, padOp.getConstantPaddingValue(), outputBuffer, storeIndices);
+      nestedBuilder.create<scf::YieldOp>(ifLoc);
+    };
+    b.create<scf::IfOp>(loopLoc, writeCond, thenBuilder);
+  };
+
+  buildNestedDistributionLoops(rewriter, loc, 0, lbs, ubs, distConfigs,
+                               innerLoopBuilder);
```

Comment:
style nit: add argument comment for better readability, especially when there are many function arguments and it is not obvious.

 ```suggestion
  buildNestedDistributionLoops(rewriter, loc, /*distributionLevel=*/0, lbs, ubs, distConfigs,
                               innerLoopBuilder);
```